### ISSUE TITLE
A bug with using Cinnamon and Coverflow (3D) OR Timeline (3D) on multiple screens

### ISSUE THREAD

**Issue by 86ul** [linuxmint/cinnamon#9993]
* Cinnamon version (cinnamon --version): 4.8.6
 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...): Mint 20.1
 * Graphics hardware *and* driver used: AMD Radeon 5500
 * 32 or 64 bit: 64b
 * Extra: A selection of inxi -Fxxxrz run in terminal:
**System**:
  Kernel: 5.4.0-67-generic x86_64 bits: 64 compiler: gcc v: 9.3.0 Desktop: Cinnamon 4.8.6 wm: muffin 4.8.1 dm: LightDM 1.30.0 Distro: Linux Mint 20.1 Ulyssa base: Ubuntu 20.04 focal 
**Machine**:
  Type: Desktop System: Gigabyte product: X570 AORUS PRO v: -CF serial: <filter> Mobo: Gigabyte model: X570 AORUS PRO v: x.x serial: <filter> UEFI: American Megatrends v: F32 date: 01/18/2021 
**CPU**:
  Topology: 8-Core model: AMD Ryzen 7 3700X bits: 64 type: MT MCP arch: Zen L2 cache: 4096 KiB flags: avx avx2 lm nx pae sse sse2 sse3 sse4_1 sse4_2 sse4a ssse3 svm bogomips: 114985 Speed: 2158 MHz min/max: 2200/3600 MHz boost: enabled Core speeds (MHz): 1: 2158 2: 2158 3: 4314 4: 2158 5: 2195 6: 2195 7: 2196 8: 2196 9: 4316 10: 2158 11: 2196 12: 2196 13: 2196 14: 2196 15: 2158 16: 2158 
**Graphics**:
  Device-1: AMD Navi 14 [Radeon RX 5500/5500M / Pro 5500M] vendor: Gigabyte driver: amdgpu v: kernel bus ID: 0a:00.0 chip ID: 1002:7340 Display: x11 server: X.Org 1.20.9 driver: amdgpu,ati unloaded: fbdev,modesetting,radeon,vesa resolution: 3440x1440 75Hz, 3440x1440 75Hz, 1920x1080 60Hz OpenGL: renderer: Radeon RX 5500 XT (NAVI14 DRM 3.35.0 5.4.0-67-generic LLVM 11.0.0) v: 4.6 Mesa 20.2.6 direct render: Yes 

**Issue**
Since a few months I have three screens, a '24 HD on the left, and 2x 32' WQHD in the middle and right. Whereas my '24 in lower than my other two '34 screens. When using Coverflow (3D) OR Timeline (3D)* to switch between the apps that are open, I see the wallpaper from my middle screen stretch across the two other screens, with space left on my left screen (the lower part, because I have set this '24 lower than the other two '34) and space left on my right screen. Thus by using Coverflow/Timeline, it use my wallpaper of my screen in middle and stretch it out over all three screens. 
I have it enabled by doing the following: Start menu > Windows > Alt-Tab > Coverflow (3D) OR Timeline (3D)

**Steps to reproduce**
Using Coverflow (3D) OR Timeline (3D) and start switching between the open apps. 

**Expected behaviour**
That Coverflow/Timeline do not stretch my wallpapers, but is just hovering over it, when using it.

**Comment by 86ul** [comment]
Same problem with Timeline (3D).

Thus Start menu > Windows > Alt-Tab > Alt-Tab switcher style > Coverflow 3D AND Timeline 3D have exactly the same bug...

**Comment by tusktenon** [comment]
I get similar visual glitches when using the Coverflow or Timeline switchers with multiple monitors, and my setup is rather different:

* Cinnamon version: 5.0.5
 * Distribution: Arch
 * Graphics hardware: Intel  TigerLake-LP GT2 [Iris Xe Graphics]
 * Graphics driver: i915 (kernel/modesetting)
 * 64 bit
 
I have an external monitor set up directly above my laptop. In my case, it's not just the wallpaper, but all of the screen contents that get messed up, along with flickering in the upper half of the external monitor. I've found a couple of (not very satisfactory) workarounds:

1. For some reason, moving the mouse cursor into the top left or top right corner of the upper display clears everything up.
2. One of the few things 86ul and I have in common with our setups is monitors at differing heights, so I went into _System Settings > Display > Layout_ and set the two screens side by side. No more glitches! For this to work, I have to keep the laptop screen (the smaller of the two) completely within the vertical boundaries of the external display; if it "sticks out" above or below, even a little, the glitches return.

**Comment by itAsia** [comment]
Same here. 

System:
  Kernel: 6.8.0-50-generic arch: x86_64 bits: 64 compiler: gcc v: 13.3.0
  Console: pty pts/2 Distro: Linux Mint 22 Wilma base: Ubuntu 24.04 noble
Machine:
  Type: Desktop System: MSI product: MS-7971 v: 1.0 serial: N/A
  Mobo: MSI model: Z170-A PRO (MS-7971) v: 1.0 serial: <filter> UEFI: American Megatrends
    v: 1.K0 date: 07/10/2018
Battery:
  Device-1: hidpp_battery_0 model: Logitech M720 Triathlon Multi-Device Mouse
    charge: 10% (should be ignored) status: discharging
  Device-2: hidpp_battery_1 model: Logitech K850 Performance Wireless Keyboard charge: 100%
    (should be ignored) status: discharging
CPU:
  Info: quad core model: Intel Core i7-6700 bits: 64 type: MT MCP arch: Skylake-S rev: 3 cache:
    L1: 256 KiB L2: 1024 KiB L3: 8 MiB
  Speed (MHz): avg: 1300 min/max: 800/4000 cores: 1: 1300 2: 1300 3: 1300 4: 1300 5: 1300
    6: 1300 7: 1300 8: 1300 bogomips: 54398
  Flags: avx avx2 ht lm nx pae sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx
Graphics:
  Device-1: NVIDIA TU116 [GeForce GTX 1660 Ti] vendor: ASUSTeK driver: nvidia v: 550.120
    arch: Turing bus-ID: 01:00.0
  Device-2: Sunplus Innovation USB Camera driver: snd-usb-audio,uvcvideo type: USB
    bus-ID: 1-10:3
  Display: x11 server: X.org v: 1.21.1.11 with: Xwayland v: 23.2.6 driver: X: loaded: nouveau
    unloaded: fbdev,modesetting,vesa failed: nvidia gpu: nvidia,nvidia-nvswitch tty: 199x47
    resolution: 1: 1920x1080 2: 3840x2160
  API: EGL v: 1.5 drivers: kms_swrast,nvidia,swrast platforms: active: gbm,surfaceless,device
    inactive: wayland,x11,device-1
  API: OpenGL v: 4.6.0 compat-v: 4.5 vendor: mesa v: 24.0.9-0ubuntu0.3
    note: console (EGL sourced) renderer: llvmpipe (LLVM 17.0.6 256 bits), NVIDIA GeForce GTX 1660
    Ti/PCIe/SSE2
Audio:
  Device-1: Intel 100 Series/C230 Series Family HD Audio vendor: Micro-Star MSI
    driver: snd_hda_intel v: kernel bus-ID: 00:1f.3
  Device-2: NVIDIA TU116 High Definition Audio vendor: ASUSTeK driver: snd_hda_intel v: kernel
    bus-ID: 01:00.1
  Device-3: Sunplus Innovation USB Camera driver: snd-usb-audio,uvcvideo type: USB
    bus-ID: 1-10:3
  Device-4: Texas Instruments PCM2902 Audio Codec driver: hid-generic,snd-usb-audio,usbhid
    type: USB bus-ID: 1-9:2
  API: ALSA v: k6.8.0-50-generic status: kernel-api
  Server-1: PipeWire v: 1.0.5 status: n/a (root, process)
Network:
  Device-1: Realtek RTL8111/8168/8211/8411 PCI Express Gigabit Ethernet vendor: Micro-Star MSI
    RTL8111/8168/8411 driver: r8169 v: kernel port: d000 bus-ID: 03:00.0
  IF: enp3s0 state: up speed: 1000 Mbps duplex: full mac: <filter>
  IF-ID-1: wg0-mullvad state: unknown speed: N/A duplex: N/A mac: N/A
Drives:
  Local Storage: total: 2.29 TiB used: 1.34 TiB (58.4%)
  ID-1: /dev/sda vendor: Crucial model: CT2000MX500SSD1 size: 1.82 TiB
  ID-2: /dev/sdb model: P3-512 size: 476.94 GiB
Partition:
  ID-1: / size: 467.89 GiB used: 226.27 GiB (48.4%) fs: ext4 dev: /dev/sdb2
  ID-2: /boot/efi size: 511 MiB used: 6.1 MiB (1.2%) fs: vfat dev: /dev/sdb1
Swap:
  ID-1: swap-1 type: file size: 2 GiB used: 35.8 MiB (1.7%) file: /swapfile
Sensors:
  System Temperatures: cpu: 67.0 C pch: 41.0 C mobo: N/A gpu: nvidia temp: 48 C
  Fan Speeds (rpm): N/A
Info:
  Memory: total: 64 GiB available: 62.76 GiB used: 31.48 GiB (50.2%)
  Processes: 466 Uptime: 8d 4h 41m Init: systemd target: graphical (5)
  Packages: 2381 Compilers: gcc: 13.3.0 Shell: Bash v: 5.2.21 inxi: 3.3.34

**Comment by gfsheppard** [comment]
Given that this is an ongoing problem, and not likely to be fixed, there should at the very least be a warning that the use of Coverflow 3D and Timeline 3D with multiple monitors will cause monitors to freeze.
I recently installed Mint 22.1, and while playing with windows settings, I thought I would try this, as it is a neat effect. 
Several monitor freezes later, I checked online to see that this is a **common** problem.
I can open a new feature request, but I thought I would post this here
